test/models/user_test.rb

```diff
@@ -335,4 +335,204 @@ def test_notes_template_validation
+  end
+
+  def test_delete_observations
+    assert_operator(0, "<", Observation.where(user: rolf).count)
+    rolf.delete_observations
+    assert_equal(0, Observation.where(user: rolf).count)
+  end
+
+  def test_delete_private_name_descriptions
+    # All Rolf's descriptions should be "private" but these two:
+    # Created by rolf, but katrina is author and editor.
+    desc1 = name_descriptions(:coprinus_desc)
+    # Created and authored by rolf, but mary is also editor.
+    desc2 = name_descriptions(:coprinus_comatus_desc)
+
+    assert(NameDescription.where(user: rolf).count > 2)
+    rolf.delete_private_name_descriptions
+    assert_obj_list_equal(NameDescription.where(user: rolf).to_a,
+                          [desc1, desc2])
+
+    # All of Dick's should be deletable, and make sure all versions of
+    # Peltigera are also deleted.
+    desc3 = name_descriptions(:peltigera_desc)
+    versions = NameDescription::Version.where(name_description_id: desc3.id)
+    assert(versions.count > 1)
+    assert_operator(0, "<", NameDescription.where(user: dick).count)
+    dick.delete_private_name_descriptions
+    assert_equal(0, NameDescription.where(user: dick).count)
+    versions = NameDescription::Version.where(name_description_id: desc3.id)
+    assert_equal(0, versions.count)
+  end
+
+  def test_delete_private_location_descriptions
+    albion = location_descriptions(:albion_desc)
+    albion.editors << dick
+    assert_users_equal(rolf, albion.user)
+    assert(albion.versions.count > 1)
+
+    # Can't delete while Dick is an editor.
+    rolf.delete_private_location_descriptions
+    assert_not_nil(LocationDescription.find(albion.id))
+
+    # Can delete if we take Dick off.
+    albion.editors.clear
+    rolf.delete_private_location_descriptions
+    assert_raises(ActiveRecord::RecordNotFound) \
+      { LocationDescription.find(albion.id) }
+    assert_equal(0, LocationDescription::Version.where(
+                  location_description_id: albion.id
+                ).count)
+  end
+
+  def test_delete_private_projects
+    # Dick created this (and several other) projects, but this one has
+    # a bunch of admins and members, so it should not be deletable.
+    bolete = projects(:bolete_project)
+    assert_users_equal(dick, bolete.user)
+    deleteable_proj_count = Project.joins(:admin_group, :user_group).
+                            where(user: dick,
+                                  admin_group: { name: "user #{dick.id}" },
+                                  user_group: { name: "user #{dick.id}" }).
+                            count
+    assert(deleteable_proj_count >= 1,
+           "user should have >= 1 project that's deleted when s/he's deleted")
+    old_proj_count = Project.where(user: dick).count
+
+    dick.delete_private_projects
+
+    assert(Project.where(user: dick).count ==
+             old_proj_count - deleteable_proj_count)
+    assert_not_empty(Project.where(id: bolete.id))
```

Comment:
```ruby
assert_not_empty(Project.where(id: bolete.id))`
``` 
instead of 
```ruby 
assert_not_nil(Project.find(bolete.id))
```
because the latter throws `ActiveRecord::RecordNotFound` if the Project doesn't exist.